incubator/hnc/e2e/tests/issues_test.go

```diff
@@ -36,4 +42,71 @@ var _ = Describe("Issues", func() {
 		runShouldContain("No conditions", 1, "kubectl hns describe", nsParent)
 		runShouldContain("No conditions", 1, "kubectl hns describe", nsChild)
 	})
+
+	It("Should set SubnamespaceAnchorMissing condition if the anchor is missing - issue #501", func() {
+		// Setting up a 3-level tree with 'parent' as the root
+		mustRun("kubectl create ns", nsParent)
+		// create a subnamespace without anchor by creating a full namespace with SubnamespaceOf annotation 
+		mustRun("kubectl create ns", nsSub1)
+		mustRun("kubectl hns set", nsSub1, "--parent", nsParent)
+		mustRun("kubectl annotate ns", nsSub1, "hnc.x-k8s.io/subnamespaceOf=" + nsParent)
+		// If the subnamespace doesn't allow cascadingDelete and the anchor is missing in the parent namespace, it should have 'SubnamespaceAnchorMissing' condition while its descendants shoudn't have any conditions."
+		// Expected: 'sub1' namespace is not deleted and should have 'SubnamespaceAnchorMissing' condition; no other conditions."
+		runShouldContain("SubnamespaceAnchorMissing", 1, "kubectl get hierarchyconfigurations.hnc.x-k8s.io -n", nsSub1, "-o yaml")
+	})
+
+	It("Should unset SubnamespaceAnchorMissing condition if the anchor is read - issue #501", func(){
+		// set up
+		mustRun("kubectl create ns", nsParent)
+		// create a subnamespace without anchor by creating a full namespace with SubnamespaceOf annotation 
+		mustRun("kubectl create ns", nsSub1)
+		mustRun("kubectl hns set", nsSub1, "--parent", nsParent)
+		mustRun("kubectl annotate ns", nsSub1, "hnc.x-k8s.io/subnamespaceOf=" + nsParent)
+		// If the anchor is read, it should unset the 'SubnamespaceAnchorMissing' condition in the subnamespace.
+		// Operation: recreate the 'sub1' subns in 'parent' - kubectl hns create sub1 -n parent
+		// Expected: no conditions.
+		mustRun("kubectl hns create", nsSub1, "-n", nsParent)
```

Comment:
Add a `runShouldContain("SubnamespaceAnchorMissing",...` before adding the anchor back to show that the condition was there but then gone after the anchor is added.

Reply:
It would cover the first test case `Should set SubnamespaceAnchorMissing condition if the anchor is missing` in this circumstance. Should we just merge these two tests then?

Reply:
It would be nice to have each block testing 1 thing but it also makes the test time longer. I'm fine with either. @adrianludwin WDYT?

Reply:
I'd prioritize test readability over test run speed for now.

Reply:
Um it looks confusing to me that there are separate two tests with exact the same code but the latter test has one more line at the end

Reply:
i.e. the way it is now